exercises/practice/dnd-character/.meta/tests.toml

```diff
@@ -0,0 +1,73 @@
+# This is an auto-generated file.
+#
+# Regenerating this file via `configlet sync` will:
+# - Recreate every `description` key/value pair
+# - Recreate every `reimplements` key/value pair, where they exist in problem-specifications
+# - Remove any `include = true` key/value pair (an omitted `include` key implies inclusion)
+# - Preserve any other key/value pair
+#
+# As user-added comments (using the # character) will be removed when this file
+# is regenerated, comments can be added via a `comment` key.
+
+[1e9ae1dc-35bd-43ba-aa08-e4b94c20fa37]
+description = "ability modifier -> ability modifier for score 3 is -4"
+
+[cc9bb24e-56b8-4e9e-989d-a0d1a29ebb9c]
+description = "ability modifier -> ability modifier for score 4 is -3"
+
+[5b519fcd-6946-41ee-91fe-34b4f9808326]
+description = "ability modifier -> ability modifier for score 5 is -3"
+
+[dc2913bd-6d7a-402e-b1e2-6d568b1cbe21]
+description = "ability modifier -> ability modifier for score 6 is -2"
+
+[099440f5-0d66-4b1a-8a10-8f3a03cc499f]
+description = "ability modifier -> ability modifier for score 7 is -2"
+
+[cfda6e5c-3489-42f0-b22b-4acb47084df0]
+description = "ability modifier -> ability modifier for score 8 is -1"
+
+[c70f0507-fa7e-4228-8463-858bfbba1754]
+description = "ability modifier -> ability modifier for score 9 is -1"
+
+[6f4e6c88-1cd9-46a0-92b8-db4a99b372f7]
+description = "ability modifier -> ability modifier for score 10 is 0"
+
+[e00d9e5c-63c8-413f-879d-cd9be9697097]
+description = "ability modifier -> ability modifier for score 11 is 0"
+
+[eea06f3c-8de0-45e7-9d9d-b8cab4179715]
+description = "ability modifier -> ability modifier for score 12 is +1"
+
+[9c51f6be-db72-4af7-92ac-b293a02c0dcd]
+description = "ability modifier -> ability modifier for score 13 is +1"
+
+[94053a5d-53b6-4efc-b669-a8b5098f7762]
+description = "ability modifier -> ability modifier for score 14 is +2"
+
+[8c33e7ca-3f9f-4820-8ab3-65f2c9e2f0e2]
+description = "ability modifier -> ability modifier for score 15 is +2"
+
+[c3ec871e-1791-44d0-b3cc-77e5fb4cd33d]
+description = "ability modifier -> ability modifier for score 16 is +3"
+
+[3d053cee-2888-4616-b9fd-602a3b1efff4]
+description = "ability modifier -> ability modifier for score 17 is +3"
+
+[bafd997a-e852-4e56-9f65-14b60261faee]
+description = "ability modifier -> ability modifier for score 18 is +4"
+
+[4f28f19c-2e47-4453-a46a-c0d365259c14]
+description = "random ability is within range"
+
+[385d7e72-864f-4e88-8279-81a7d75b04ad]
+description = "random character is valid"
+
+[2ca77b9b-c099-46c3-a02c-0d0f68ffa0fe]
+description = "each ability is only calculated once"
+include = false
+
+[dca2b2ec-f729-4551-84b9-078876bb4808]
+description = "each ability is only calculated once"
+reimplements = "2ca77b9b-c099-46c3-a02c-0d0f68ffa0fe"
+include = false
```

Comment:
Are neither of these to be used?

Reply:
I think that because the character is returned as a struct this test would be meaningless. There's no way for an ability to change after the fact like there would be if it were an opaque object with abilities accessible only via a getter.

Reply:
The second is a new version of the first, so we're not supposed to implement the first either way. But in general those tests don't make much sense in C. We return a struct, so there's no way things are computed multiple times. The C++ [implementation](https://github.com/exercism/cpp/tree/main/exercises/practice/dnd-character) uses plain attributes as well, although there we could have methods. In a language supporting actual properties like Python it makes more sense, and they do [implement](https://github.com/exercism/python/blob/main/exercises/practice/dnd-character/dnd_character_test.py#L76) those tests as well.